libvast/src/system/signal_monitor.cpp

```diff
@@ -41,24 +41,30 @@ extern "C" void signal_monitor_handler(int sig) {
   }
   signals[0] = true;
   signals[sig] = true;
+  cvp->notify_one();
```

Comment:
The standard says:

> The behavior of any function other than a [plain old function] used as a signal handler in a C++ program is implementation-defined.

The function `notify_one` is not from a "common subset", so the code is not well defined.

Reply:
TIL! I dismissed my approval.

I found this great blog post on an approach that is not UB from what I can tell. This works by blocking signals in all threads except for the signal monitor one, and then having the signal monitor thread use `sigwait` rather than a condition variable.

https://thomastrapp.com/posts/signal-handlers-for-multithreaded-c++/#no-need-to-call-me-ill-sigwait-for-you

Reply:
I half expected that there was a catch. Well at least @dominiklohmann already found a blueprint for a more sound approach.